### PR TITLE
Add API docs

### PR DESCRIPTION
**Pull request by juankuquintana**
Belongs to https://github.com/dnsimple/dnsimple-business/issues/1662.

This PR is a proposal of what API changes we will have to do to support async registrations, renewals, and transfers. As we are now treating registrations, renewals, and transfers as asynchronous, our API users should have a way to know what the status of an operation is.

This PR adds the following endpoints:

- [getDomainRegistration](http://localhost:3000/v2/registrar/#getDomainRegistration)
- [getDomainRenewal](http://localhost:3000/v2/registrar/#getDomainRenewal)

_We don't have any new endpoints for a domain transfer as we support this already._

<!--
## :shipit: Verification

- [ ] Docs are updated correctly

### CODE REVIEW

File: content/v2/registrar.markdown [hunk 1]
```diff
@@ -230,6 +230,44 @@ Responds with [HTTP 401](/v2#unauthorized) in case of case of authentication iss
...
+
+Name | Type | Description
+-----|------|------------
+`:account` | `integer` | The account id
+`:domain` | `string` | The domain name
+`:domain_registration` | `integer` | The domain registration id
+
+### Example
+
+Get the domain registration with ID `361` in the account `1010`:
```
Comment (DXTimer) [inline content/v2/registrar.markdown:251]: Missing mention of the domain parameter.

File: content/v2/openapi.yml [hunk 2]
```diff
@@ -952,6 +952,30 @@ paths:
...
+      description: Retrieves the details of an existing domain registration.
+      summary: Retrieve a domain registration
+      parameters:
+        - $ref: '#/components/parameters/Account'
+        - $ref: '#/components/parameters/Domain'
+        - $ref: '#/components/parameters/DomainRegistration'
+      operationId: getDomainRegistration
+      tags:
+        - registrar
+        - registrar registration
```
Comment (DXTimer) [inline content/v2/openapi.yml:966]: Would need to add this tag to the global tags.

File: fixtures/v2/api/getDomainRegistration/success.http [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+X-Request-Id: 9f4959ee-06a9-488c-906e-27a570cafbbf
+X-Runtime: 0.078429
+X-Frame-Options: DENY
+X-Content-Type-Options: nosniff
+X-XSS-Protection: 1; mode=block
+X-Download-Options: noopen
+X-Permitted-Cross-Domain-Policies: none
+Strict-Transport-Security: max-age=31536000
+
+{"data":{"id":361,"domain_id":104040,"registrant_id":2715,"period":1,"state":"registering","auto_renew":false,"whois_privacy":false,"estimated_completion_at":"2023-02-01T17:44:32Z","created_at":"2023-01-27T17:44:32Z","updated_at":"2023-01-27T17:44:40Z"}}
```
Comment (DXTimer) [inline fixtures/v2/api/getDomainRegistration/success.http:20]: The `estimated_completion_at` attribute would need to be added to the component schemas of the `DomainRegistration`, `DomainRenewal`, and `DomainTransfer`.

**Review by DXTimer — APPROVED** [review]
Looks great.

**Comment by OleMchls** [comment]
Verified on dnsimple.dev 👌

**Comment by weppos** [comment]
@DXTimer @juankuquintana any chance we can document the enum of states that someone could expect from a DomainRegistration, DomainRenewal, and DomainTransfer.

I see a fixture with includes and example with "registering". As a consumer, I may want to know what other state strings to perpare for.

**Comment by juankuquintana** [comment]
> @DXTimer @juankuquintana any chance we can document the enum of states

@weppos Took a quick look at our dev docs and did not found an example for documenting the response objects (please point me if we have done this), i.e. this screenshot from Stripe

Screen Shot 2023-02-23 at 13 37 23

We could go as far as start documenting the response objects, small like adding a line of text after the response describing how the state behaves, or even modify the response to `"state": "registering|registered|cancelled" ` could be understood by devs.

What would be an accepted and preferable way to solve this?

**Comment by DXTimer** [comment]
@juankuquintana we could use an info admonition box after the response example, and we can update the OpenAPI Spec with something similar to:

```
        state:
          type: string
          description: The state of the registration
          enum:
            - new
            - registering
            - registered
            - failed
            - cancelling
            - cancelled
```

**Pull request merged**